src/main/java/lesson5/task5/Bicycle.java

```diff
@@ -0,0 +1,29 @@
+package lesson5.task5;
+
+public class Bicycle extends Vehicle {
+
+    public Bicycle(boolean engineStarted) {
+        super(engineStarted);
+    }
+
+
+    @Override
+    protected void startEngine() {
+        throw new UnsupportedOperationException("Bicycle does not have engine");
```

Comment:
This breakes SOLID rules (L - Liskov Substitution Principle) when you will call startEngine() on Bicycle your program will throw exception. Since Bicycles normally don't have engines, why to extend a interface/class that forces us to implement engine there.

Reply:
Adjusted. Please review.